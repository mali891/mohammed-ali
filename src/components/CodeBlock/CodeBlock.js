import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import Prism from 'prismjs'

import { styles } from './CodeBlock.styles'

const CodeBlock = ({ children, language }) => {
  useEffect(() => Prism.highlightAll(), [])

  return (
    <pre css={styles}>
      <code className={`language-${language}`}>{children}</code>
    </pre>
  )
}

CodeBlock.defaultProps = {
  language: 'javascript'
}

CodeBlock.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  language: PropTypes.string
}

export default CodeBlock
