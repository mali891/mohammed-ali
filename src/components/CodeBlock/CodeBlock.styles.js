import { css } from '@emotion/core'
import { COLOURS } from '../../styles/settings'
import '../../styles/libs/prism.css'

export const styles = css`
  background-color: ${COLOURS.BLACK};
`
