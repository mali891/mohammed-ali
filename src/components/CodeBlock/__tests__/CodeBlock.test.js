import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import CodeBlock from '../CodeBlock'

describe('CodeBlock', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <CodeBlock>
        <p>Dummy children</p>
      </CodeBlock>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given an alternative language prop', () => {
    const { asFragment, container } = render(
      <CodeBlock language="scss">
        <p>Dummy children</p>
      </CodeBlock>
    )

    expect(container.querySelector('.language-scss')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
