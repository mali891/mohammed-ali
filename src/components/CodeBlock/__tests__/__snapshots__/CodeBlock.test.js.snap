// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock renders correctly 1`] = `
<DocumentFragment>
  <section
    class="c-content-group"
    css="[object Object]"
  >
    <pre
      class=" language-javascript"
      css="[object Object]"
    >
      <code
        class=" language-javascript"
      >
        Dummy children
      </code>
    </pre>
  </section>
</DocumentFragment>
`;

exports[`CodeBlock renders correctly given an alternative language prop 1`] = `
<DocumentFragment>
  <section
    class="c-content-group"
    css="[object Object]"
  >
    <pre
      class=" language-scss"
      css="[object Object]"
    >
      <code
        class=" language-scss"
      >
        Dummy children
      </code>
    </pre>
  </section>
</DocumentFragment>
`;
